de.metas.printing/de.metas.printing.base/src/main/java/de/metas/printing/async/spi/impl/PDFDocPrintingWorkpackageProcessor.java

```diff
@@ -0,0 +1,241 @@
+
+import de.metas.adempiere.report.jasper.OutputType;
+import de.metas.adempiere.report.jasper.client.JRClient;
+import de.metas.async.Async_Constants;
+import de.metas.async.api.IQueueDAO;
+import de.metas.async.model.I_C_Async_Batch;
+import de.metas.async.model.I_C_Queue_WorkPackage;
+import de.metas.async.spi.IWorkpackageProcessor;
+import de.metas.i18n.ILanguageBL;
+import de.metas.i18n.IMsgBL;
+import de.metas.i18n.Language;
+import de.metas.printing.MergePdfByteArrays;
+import de.metas.printing.PrintPackagePDFBuilder;
+import de.metas.printing.api.IPrintPackageBL;
+import de.metas.printing.api.IPrintingDAO;
+import de.metas.printing.api.impl.PrintPackageCtx;
+import de.metas.printing.model.I_C_Print_Job_Instructions;
+import de.metas.printing.model.I_C_Print_Job_Line;
+import de.metas.printing.model.I_C_Print_Package;
+import de.metas.printing.model.I_C_Printing_Queue;
+import de.metas.printing.model.X_C_Print_Job_Instructions;
+import de.metas.process.IADPInstanceDAO;
+import de.metas.process.ProcessInfo;
+import de.metas.process.ProcessInfoParameter;
+
+/**
+ * @author cg
+ * 
+ */
+public class PDFDocPrintingWorkpackageProcessor implements IWorkpackageProcessor
+{
+	// services
+	private final IPrintingDAO dao = Services.get(IPrintingDAO.class);
+	private final IQueueDAO queueDAO = Services.get(IQueueDAO.class);
+
+	public static final int SummaryPdfPrinting_AD_Process_ID = 540661;
+
+	private I_C_Async_Batch asyncBatch;
+
+	@Override
+	public Result processWorkPackage(final I_C_Queue_WorkPackage workpackage, final String localTrxName) 
+	{
+		this.asyncBatch = workpackage.getC_Async_Batch();
+		Check.assumeNotNull(asyncBatch, "Async batch is not null");
+
+		final List<I_C_Print_Job_Instructions> list = queueDAO.retrieveItems(workpackage, I_C_Print_Job_Instructions.class, localTrxName);
+		for (final I_C_Print_Job_Instructions printjobInstructions : list)
+		{
+			InterfaceWrapperHelper.refresh(printjobInstructions, localTrxName);
+			createPrintPackage(printjobInstructions, localTrxName);
+			if (X_C_Print_Job_Instructions.STATUS_Error.equals(printjobInstructions.getStatus()))
+			{
+				throw new AdempiereException(printjobInstructions.getErrorMsg());
+			}
+
+			try
+			{
+				print(printjobInstructions, workpackage, localTrxName);
+			}
+			catch (Exception e)
+			{
+				throw new AdempiereException(e.getMessage());
+			}
+		}
+		return Result.SUCCESS;
+	}
+
+	private void print(final I_C_Print_Job_Instructions jobInstructions, final I_C_Queue_WorkPackage workpackage, final String trxName) throws Exception
+	{
+		//
+		// Extract print packages
+		int countLines = 0;
+		final Iterator<I_C_Print_Job_Line> jobLines = dao.retrievePrintJobLines(jobInstructions);
+		final Map<Integer, I_C_Print_Package> printPackages = new LinkedHashMap<Integer, I_C_Print_Package>();
+
+		boolean isCreateSummary = false; // TODO : needs to be on true in order to have summary which currently in not working FIXME in a next working increment
```

Comment:
please add the follow-up issues URL..otherwise this TODO might be around forever..